src/main/java/com/ibm/cloud/cloudant/kafka/transforms/ArrayFlatten.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.kafka.transforms;
```

Comment:
This is a more general discussion point, since the original package names were quite poor.

Do we want to have some sort of internal/external split in package namespaces? In particular so that we can signpost that the transforms are for "public consumption" but other conversions and utilities aren't.

Reply:
This is an interesting point. I suppose the thing with Kafka Connect is that a user is never expected to interact with the classes in a Java sense since everything is via configuration so the package name shows up in those connection strings, but not elsewhere and there is no need for published JavaDoc because it doesn't make sense in this context (only the `ConfigDef` - which would be nice to trasnform into docs thinking about it).

That said we could still make it more obvious which classes are expected to have their names appear "publicly" in config strings. We could put the `CloudantSourceConnector` `CloudantSinkConnector` in 1 package.
The 3 transforms (schema, array flatten, and omit design docs) in another package (or sub package of the first).
And everything else in an "internal" package.

Thoughts?

Reply:
This seems like a reasonable package design.  +1 for having the transforms by themselves and not intermingled with the other classes.